artiq/master/experiments.py

```diff
@@ -191,6 +193,8 @@ def dispose(self):
 
 class GitBackend:
     def __init__(self, root):
+        self.root = os.path.abspath(root)
```

Comment:
This is ok.

You may also want to differentiate between workdir and root a bit more and catch the case of a bare repo. Add a `workdir()` method to both backends, relying on the normalization by pygit2:

```ipython
In [2]: pygit2.Repository("/tmp/r").workdir
Out[2]: '/tmp/r/'

In [3]: pygit2.Repository("/tmp/r").path
Out[3]: '/tmp/r/.git/'

In [4]: pygit2.Repository("/tmp/r/.git").path
Out[4]: '/tmp/r/.git/'

In [5]: pygit2.Repository("/tmp/r/.git").workdir
Out[5]: '/tmp/r/'

In [6]: pygit2.init_repository("/tmp/rb", bare=True)
Out[6]: pygit2.Repository('/tmp/rb/')

In [8]: pygit2.Repository("/tmp/rb").workdir

In [9]: pygit2.Repository("/tmp/rb").path
Out[9]: '/tmp/rb/'
```